src/utilities/process-readme.js

```diff
@@ -1,6 +1,70 @@
 const url = require('url');
 
-const beginsWithDocsDomainRegex = /^https?:\/\/webpack\.js\.org/;
+const beginsWithDocsDomainRegex = /^(?:https?:)\/\/webpack\.js\.org/;
+const inlineLinkRegex = /\[[^\]]*\]\(([^)]+)\)/g;
+const externalLinkRegex = /^[ \t]*\[[^\]]+\]:[ \t]+([^ \n]+)/gm;
+
+const fragmentLinkMap = {
+  '/api/module-variables/#__webpack_public_path__-webpack-specific-': '/api/module-variables/#__webpack_public_path__-webpack-specific',
+  '/configuration/module/#rule-exclude': '/configuration/module/#ruleexclude',
+  '/configuration/module/#rule-include': '/configuration/module/#ruleinclude',
+  '/configuration/module/#rule-options-rule-query': '/configuration/module/#ruleoptions--rulequery',
+  '/configuration/module/#rule-use': '/configuration/module/#ruleuse',
+  '/configuration/optimization/#optimization-concatenatemodules': '/configuration/optimization/#optimizationconcatenatemodules',
+  '/configuration/output/#output-chunkfilename': '/configuration/output/#outputchunkfilename',
+  '/configuration/output/#output-publicpath': '/configuration/output/#outputpublicpath',
+  '/configuration/resolve/#resolve-modules': '/configuration/resolve/#resolvemodules',
+  '/guides/shimming/#exports-loader': '/loaders/exports-loader',
+  '/guides/shimming/#imports-loader': '/loaders/imports-loader',
+  '/guides/shimming/#provideplugin': '/plugins/provide-plugin/',
+};
+
+function linkFixerFactory(sourceUrl) {
+  return function linkFixer(markdownLink, href) {
+    const oldHref = href;
+
+    if (href.includes('//npmjs.com')) {
+      href = href.replace('//www.npmjs.com');
+    }
+
+    // Only resolve non-absolute urls from their source if they are not a document fragment link
+    if (!href.startsWith('#')) {
+      // Convert Github raw links to rendered links
+      let rendered_url = sourceUrl
+        .replace(/raw.githubusercontent.com/, 'github.com')
+        .replace(/master/, 'blob/master');
+
+      href = url.resolve(rendered_url, href);
+    }
+
+    // Modify absolute documenation links to be root relative
+    if (beginsWithDocsDomainRegex.test(href)) {
+      href = href.replace(beginsWithDocsDomainRegex, '');
+    }
+
+    const fragmentLinkMapMatch = Object.keys(fragmentLinkMap).find(mapFrom => href.includes(mapFrom));
+    if (fragmentLinkMapMatch) {
+      href = href.replace(fragmentLinkMapMatch, fragmentLinkMap[fragmentLinkMapMatch]);
+      console.error(`DEPRECATED EXTERNAL README LINK:
```

Comment:
or here? (Can this include the faulty readme file source so we know where to put a PR to fix it?)

Reply:
The url of the readme is shown here: https://github.com/webpack/webpack.js.org/pull/3265/files/3de50b8b70331e4943e8c9f6022069001378b7ce#diff-de191ad0d27673dbfaf4f71e79e397c0R49

See output here: https://travis-ci.org/webpack/webpack.js.org/jobs/576537198#L294-L297

Maybe I should make that a clickable edit link instead, to help us out a bit

Reply:
Would actually be nice to pipe those errors into a Slack channel or something similar. No one is ever going to inspect this log

Reply:
Github allows direct edit URL's with prefilled title and description: https://github.com/webpack-contrib/bundle-loader/edit/master/README.md?message=Fix%20broken%20link&description=This%20broken%20URL%20was%20found%20by%20the%20webpack.js.org%20link%20checker

This means the workflow could be reduced to clicking a link and manually making the replacement. Maybe we can prefill even more from the outside

Reply:
thats enough, ty

Reply:
but the nice to haves are welcome at any time i suppose, they sound pretty handy